helper.go

```diff
@@ -192,13 +192,9 @@ func (s Schema) IsTrivial(refResolvers ...func(string) (SchemaOrBool, bool)) boo
 		resolved := false
 
 		for _, resolve := range refResolvers {
-			if rs, found := resolve(*s.Ref); found {
+			if _, found := resolve(*s.Ref); found {
 				resolved = true
-
-				if !rs.IsTrivial(refResolvers...) {
-					return false
-				}
-
+				// Do not traversal existed ref.
```

Comment:
Removing traversal entirely would break `IsTrivial` logic (see suggested test cases). 

So it is better to add a check for already visited references. Example implementation:
```go
		for _, resolve := range refResolvers {
			if rs, found := resolve(*s.Ref); found {
				resolved = true

				// If same ref is met, it is returned as a trivial schema to avoid duplicate recursion.
				skipRef := func(ref string) (SchemaOrBool, bool) {
					if ref == *s.Ref {
						return SchemaOrBool{}, true
					}

					return SchemaOrBool{}, false
				}

				if !rs.IsTrivial(append([]func(ref string) (SchemaOrBool, bool){skipRef}, refResolvers...)...) {
					return false
				}

				break
			}
		}
```